File: examples/python/members.py
class Bird:
    def __init__(self, color):
        self.color = color

    def fly(self):
        print("flapflapflap")

class Duck(Bird):
    pass

penguin = Bird("black and white")
parrot = Bird("green etc.")
sally = Duck("various shades of brown")

print(sally.color)

# We can just add new members, because we're
# really just putting a new value in the dictionary:

sally.favorite_ice_cream = "snail flavor"  # she’s a duck
print(sally.favorite_ice_cream)

# Methods are just property accesses too:

sally_fly_method = sally.fly
sally_fly_method()

File: examples/ruby/members.rb
class Bird
  def initialize(color)
    @color = color
  end

  def fly
    puts("flapflapflap and I am " + @color)
  end

  attr_accessor :color

  # Is sugar for:
  #
  # attr_accessor(:color)
  #
  # ...which generates:
  #
  # def color
  #   @color
  # end
  #
  # def color=(new_color)  # The actual method name is “color=” (wat?!)
  #   @color = new_color
  # end
end

class Duck < Bird
end

sally = Duck.new("various shades of brown")
sally.fly
sally.fly()  # Same thing! In Ruby, foo.bar is _always_ a method call
puts sally.color

# Note that we can’t get at another object’s instance variables.
# This is not legal syntax:
#
# sally.@color

# Class declarations are just code that runs from top to bottom.
# Let’s add 100 feather properties:

class Bird
  100.times { |n| attr_accessor("feather#{n}") }
end

sally.feather73 = "wingtip"
puts sally.feather73
